### PR TITLE
Go1.17 updates

### PR DESCRIPTION
**Pull request by ChrisHines**
See commit messages for details.

There are more dependencies that we could upgrade, but I only upgraded `github.com/go-kit/log` so far. Here's the list of possible upgrades at this moment:

```
>go-mod-upgrade
   • Discovering modules...   
? Choose which modules to update  [Use arrows to move, space to select, <right> to all, <left> to none, type to filter]
> [ ]  github.com/aws/aws-sdk-go                       1.38.68                           -> 1.40.45
  [ ]  github.com/aws/aws-sdk-go-v2                    1.7.0                             -> 1.9.0
  [ ]  github.com/aws/aws-sdk-go-v2/service/cloudwatch 1.5.0                             -> 1.8.0
  [ ]  github.com/casbin/casbin/v2                     2.31.6                            -> 2.37.0
  [ ]  github.com/hashicorp/consul/api                 1.8.1                             -> 1.10.1
  [ ]  github.com/nats-io/nats-server/v2               2.2.6                             -> 2.5.0
  [ ]  github.com/nats-io/nats.go                      1.11.0                            -> 1.12.1
  [ ]  go.uber.org/zap                                 1.17.0                            -> 1.19.1
  [ ]  golang.org/x/time                               0.0.0-20200416051211-89c76fbcd5d1 -> 0.0.0-20210723032227-1f47c861a9ac
  [ ]  google.golang.org/grpc                          1.38.0                            -> 1.40.0
```

Should we bump any of these?

### CODE REVIEW

**Review by sagikazarmark — APPROVED** [review]
LGTM

Let's update the rest of the libraries before tagging 0.12. I can take care of that if you want me to.

**Comment by ChrisHines** [comment]
It's easy enough for me to do it. I'll add another commit here soon.

**Comment by ChrisHines** [comment]
Well, now the race detector is failing `github.com/go-kit/kit/sd/eureka` integration tests. It may not be related to the dependency upgrades, though. The race seems to be between something in the `sd/eureka` package and something in the `github.com/hudl/fargo` package and as far as I can tell the last commit didn't upgrade `github.com/hudl/fargo` or any of its dependencies.

Thoughts?

**Comment by sagikazarmark** [comment]
The eureka tests are flaky. Haven't been able to figure out why, but then again, it'll probably be deprecated and removed at some point.

**Review by sagikazarmark — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ jobs:
     name: Build
     runs-on: ubuntu-latest
     strategy:
-      matrix:  # Support latest and one minor back
-        go: ['1.15', '1.16']
+      matrix: # Support latest and one minor back
+        go: ["1.16", "1.17"]
```
Comment (peterbourgon) [inline .github/workflows/ci.yml:15]: Just for the record this pulls the latest patch release for each minor, which is what we wan't but wasn't obvious to me.

**Review by peterbourgon — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ChrisHines) [reply .github/workflows/ci.yml:15]: Yes. I double checked the logs to make sure it did that because I wasn't sure either.

**Comment by sagikazarmark** [comment]
This PR was the last item in the v0.12.0 milestone. Is there anything else we want to do before a new release?

**Comment by ChrisHines** [comment]
I don't have anything else.

**Pull request merged**